src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -663,11 +663,100 @@ public void AnalyzeCommandBase_ReportsWarningOnUnsupportedPlatformForRuleAndNoRu
             return run;
         }
 
+        [Fact]
+        public void AnalyzeCommand_Traces()
+        {
+            var sb = new StringBuilder();
+
+            foreach (DefaultTraces trace in new[] { DefaultTraces.None, DefaultTraces.ScanTime, DefaultTraces.RuleScanTime })
+            {
+                var options = new TestAnalyzeOptions
+                {
+                    OutputFilePath = Guid.NewGuid().ToString(),
+                    TargetFileSpecifiers = new string[] { Guid.NewGuid().ToString() },
+                    Traces = new[] { trace.ToString() },
+                    Level = new[] { FailureLevel.Warning, FailureLevel.Note },
+                };
+
+                Run run = RunMultithreadedAnalyzeCommand(ComprehensiveKindAndLevelsByFilePath,
+                                                         generateDuplicateScanTargets: false,
+                                                         expectedResultCode: SUCCESS,
+                                                         expectedResultCount: WARNING_COUNT + NOTE_COUNT,
+                                                         options);
+
+
+                int resultCount = 0;
+                int executionNotificationsCount = 0;
+                int configurationNotificationCount = 0;
+
+                SarifHelpers.ValidateRun(
```

Comment:
>ValidateRun [](http://example.com/codeflow?start=29&length=11)

Note that this validation will halt the test early if it fails. That's ok. If this fails, then the follow-on validation is unlikely to succeed, because we already know that we don't have the expected results + notifications data generated.